web-server/v0.4/src/routes/Dashboard/RunComparison.js

```diff
@@ -334,30 +334,70 @@ export default class RunComparison extends ReactJS.Component {
     let end = this.state.checked.length;
     var position = 5;
     var fullIHeight = 0;
-    html2canvas(document.querySelector('.ant-layout-content'), { allowTaint: true }).then(function(
-      canvas
-    ) {
-      canvas.getContext('2d');
-      var imgData = canvas.toDataURL();
-      var imgWidth = 210;
-      var pageHeight = 295;
-      var imgHeight = (canvas.height * imgWidth) / canvas.width;
-      position = fullIHeight + 5;
-      fullIHeight = fullIHeight + imgHeight;
-      var heightLeft = fullIHeight;
-      doc = doc.addImage(imgData, 'PNG', 0, position, imgWidth, imgHeight);
-      heightLeft -= pageHeight;
-      while (heightLeft >= 0) {
-        position = heightLeft - fullIHeight;
-        console.log('position renew : ' + position);
-        doc.addPage();
+    var requests = this.state.checked.map((check, index) => {
+      html2canvas(document.querySelector('#' + check), { allowTaint: true }).then(function(canvas) {
+        canvas.getContext('2d');
+        var imgData = canvas.toDataURL();
+        var imgWidth = 210;
+        var pageHeight = 295;
+        var imgHeight = (canvas.height * imgWidth) / canvas.width;
+        position = fullIHeight + 5;
+        fullIHeight = fullIHeight + imgHeight;
+        var heightLeft = fullIHeight;
         doc = doc.addImage(imgData, 'PNG', 0, position, imgWidth, imgHeight);
         heightLeft -= pageHeight;
-      }
-      doc.save();
+        while (heightLeft >= 0) {
+          position = heightLeft - fullIHeight;
+          doc.addPage();
+          doc = doc.addImage(imgData, 'PNG', 0, position, imgWidth, imgHeight);
+          heightLeft -= pageHeight;
+        }
+        doc.save();
+      });
     });
   };
 
+  selectDiv = (type, cluster) => {
```

Comment:
Can you add some comments explaining what the different code paths mean?

Reply:
Yes, have added few comments in current commit.